src/main/java/com/streamr/client/protocol/control_layer/PublishRequestAdapter.java

```diff
@@ -0,0 +1,30 @@
+package com.streamr.client.protocol.control_layer;
+
+import com.squareup.moshi.JsonReader;
+import com.squareup.moshi.JsonWriter;
+import com.streamr.client.protocol.message_layer.StreamMessage;
+import com.streamr.client.protocol.message_layer.StreamMessageAdapter;
+
+import java.io.IOException;
+
+public class PublishRequestAdapter extends ControlLayerAdapter<PublishRequest> {
+
+    private static final StreamMessageAdapter streamMessageAdapter = new StreamMessageAdapter();
+
+    @Override
+    public PublishRequest fromJson(JsonReader reader) throws IOException {
+        StreamMessage streamMessage = streamMessageAdapter.fromJson(reader);
+        String sessionToken = reader.nextString();
```

Comment:
Theoretically nullable, although we don't have publicly writeable streams as of yet. Wouldn't hurt to be null-safe here.